models/classes/configurationMarkers/ConfigurationMarkers.php

```diff
@@ -85,8 +83,8 @@ private function walkReplaceMarkers(&$item): void
             return;
         }
 
-        $isSecretDefined = $this->secretsStorage->exist($matches[1] ?? '');
-        $this->printMatchNotification($isSecretDefined, $matches[1]);
+        $isSecretDefined = $_ENV[$matches[1]] ?? false;
```

Comment:
instead of accessing global vars directly here, can't we inject it as a dependency ???

Reply:
Yes, I agree with @andreluizmachado . You can just do this:

```
public function __construct(
        SerializableSecretDtoFactory $serializableFactory,
        LoggerInterface $logger,
        array $envVars = null
    ) {
        $this->serializableFactory = $serializableFactory;
        $this->logger = $logger;
        $this->envVars = $envVars ?? $_ENV;
    }
```

And here 

```suggestion
        $isSecretDefined = $this->envVars[$matches[1]] ?? false;
```

Reply:
I'm a little confused here. The whole point of this change was to remove injecting env variables in __constructor. And now we want to inject it in a different place. 
What if I use `getenv` function instead of referring to a global variable?

Reply:
The point is to remove it from the container definition, otherwise it is not dynamic anymore, but static.